tests/model_registry/factory_utils.py

```diff
@@ -0,0 +1,210 @@
+# FACTORY CLASSES
+# =============================================================================
+
+
+@dataclass
+class ModelRegistryDBConfig:
+    """Configuration for Model Registry Database resources."""
+
+    name_prefix: str = "mr-db"
+    namespace: str = "default"
+    teardown: bool = True
+    mysql_image: str = MR_DB_IMAGE_DIGEST
+    storage_size: str = "5Gi"
+    access_mode: str = "ReadWriteOnce"
+    database_name: str = MODEL_REGISTRY_DB_SECRET_STR_DATA["database-name"]
+    database_user: str = MODEL_REGISTRY_DB_SECRET_STR_DATA["database-user"]
+    database_password: str = MODEL_REGISTRY_DB_SECRET_STR_DATA["database-password"]
+    port: int = 3306
+    ssl_config: Optional[Dict[str, Any]] = None
+    labels: Optional[Dict[str, str]] = None
+    annotations: Optional[Dict[str, str]] = None
+
+
+@dataclass
+class ModelRegistryConfig:
+    """Configuration for Model Registry instance."""
+
+    name: str = "model-registry"
+    namespace: str = "default"
+    use_oauth_proxy: bool = True
+    use_istio: bool = False
+    teardown: bool = True
+    grpc_config: Optional[Dict[str, Any]] = None
+    rest_config: Optional[Dict[str, Any]] = None
+    mysql_config: Optional[ModelRegistryDBConfig] = None
+    labels: Optional[Dict[str, str]] = None
+    wait_for_conditions: bool = True
+    oauth_proxy_config: Optional[Dict[str, Any]] = None
+    istio_config: Optional[Dict[str, Any]] = None
+
+
+@dataclass
+class ModelRegistryDBBundle:
+    """Bundle containing all Model Registry DB resources."""
+
+    service: Service
+    pvc: PersistentVolumeClaim
+    secret: Secret
+    deployment: Deployment
+    config: ModelRegistryDBConfig
+
+    def cleanup(self) -> None:
+        """Clean up all resources in this bundle in reverse order of creation."""
+        # Cleanup in reverse order: Deployment -> Service -> PVC -> Secret
+        resources = [
+            ("deployment", self.deployment),
+            ("service", self.service),
+            ("pvc", self.pvc),
+            ("secret", self.secret),
+        ]
+
+        for resource_type, resource in resources:
+            try:
+                if resource and hasattr(resource, "delete"):
+                    LOGGER.info(f"Cleaning up {resource_type}: {resource.name}")
+                    resource.delete(wait=True)
+            except Exception as e:
+                LOGGER.warning(f"Failed to cleanup {resource_type} {resource.name if resource else 'unknown'}: {e}")
+
+    def get_mysql_config(self) -> Dict[str, Any]:
+        """Get MySQL configuration dictionary for Model Registry."""
+        return {
+            "host": f"{self.deployment.name}.{self.deployment.namespace}.svc.cluster.local",
+            "database": self.config.database_name,
+            "passwordSecret": {"key": "database-password", "name": self.secret.name},
+            "port": self.config.port,
+            "skipDBCreation": False,
+            "username": self.config.database_user,
+        }
```

Comment:
optional parameter is missing from the mysql config `sslRootCertificateConfigMap`